Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -315,4 +323,14 @@ void _UIApplicationShutdown() {
 extern "C" void UIApplicationMainHandleToastNotificationEvent(const char* notificationData) {
     NSString* data = Strings::IsEmpty<const char*>(notificationData) ? nil : [[NSString alloc] initWithCString:notificationData];
     [[UIApplication sharedApplication] _sendNotificationReceivedEvent:data];
+}
+
+extern "C" void UIApplicationMainHandleVoiceCommandEvent(IInspectable* voiceCommandResult) {
+    WMSSpeechRecognitionResult* speechResult = [WMSSpeechRecognitionResult createWith:voiceCommandResult];
+    [[UIApplication sharedApplication] _sendVoiceCommandReceivedEvent:speechResult];
+}
+
+extern "C" void UIApplicationMainHandleProtocolEvent(IInspectable* protocolUri) {
+    WFUri* protocolResult = [WFUri createWith:protocolUri];
```

Comment:
Would be worth double checking that createWith creates an autoreleased object and that we don't need to use StrongId or similar here.